dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java

```diff
@@ -299,11 +299,27 @@ public Contentlet findContentletForLanguage(long languageId, Identifier contentl
         try {
             return findContentletByIdentifier(contentletId.getId(), false, languageId, APILocator.systemUser(), false);
         } catch (DotContentletStateException dcs) {
-            Logger.debug(this, "No working contentlet found for language");
+            Logger.debug(this, ()->String.format("No working contentlet found for language: %d and identifier: %s ", languageId, null != contentletId ? contentletId.getId() : "Unkown"));
         }
         return null;
     }
 
+    @CloseDBIfOpened
+    @Override
+    public List<Contentlet> findContentletsForAllLanguages(final Identifier identifier) {
+        final LanguageAPI languageAPI = APILocator.getLanguageAPI();
+        final List<Language> allLanguages = languageAPI.getLanguages();
+        return allLanguages.stream().map(language -> {
+            try {
+                return findContentletForLanguage(language.getId(), identifier);
+            } catch (DotDataException | DotSecurityException e) {
+                Logger.debug(this, ()->String.format("No working contentlet found for language: %s and identifier: %s ", language.toString(), null != identifier ? identifier.getId() : "Unkown"));
+                return null;
+            }
+        }).filter(Objects::nonNull).collect(CollectionsUtils.toImmutableList());
+    }
+
+
```

Comment:
I am pretty sure we have this method in a few places and don't need to add to the already enormous contentletAPI - see:
`findAllVersions`
`getAllLanguages`